### PR TITLE
[MM-41852] - Add Start Trial to final stage of onboarding checklist

### PR DESCRIPTION
**Pull request by AGMETEOR**
#### Summary
This PR adds a CTA to open the start trial modal on the final onboarding screen of the checklist

#### Ticket Link
[MM-41852](https://mattermost.atlassian.net/browse/MM-41852)

#### Release Note

```release-note
NONE

```

### CODE REVIEW

**Comment by AGMETEOR** [comment]
Hey @neallred , how do we get the task list showing on the test servers? I already set `MM_FEATUREFLAGS_USECASEONBOARDING=true` for this one but it's still not showing

**Comment by neallred** [comment]
I see it when I log in on `https://mattermost-webapp-pr-9843.test.mattermost.cloud/acme/channels/town-square` as sysadmin now. I checked the settings with the `cloud` command and it looked like the right settings were set. Based on that, I'm guessing you followed the instructions correctly (https://community-daily.mattermost.com/private-core/pl/e5duwp1mr7n4jmokrre1xa3czo) . Maybe it took longer to update the config than the time you tried it in?

i-see-task-list


I was also looking at the wrong spinwick (your other PR) at first and didn't see it, so I reran through the steps to be double sure, then noticed I was on the wrong spinwick. So I didn't see this spinwick until I ran through the steps again myself to be sure.

File: i18n/en.json [hunk 1]
```diff
@@ -4470,6 +4471,7 @@
   "tutorial_tip.ok": "Next",
   "tutorial_tip.out": "Opt out of these tips.",
   "tutorial_tip.seen": "Seen this before? ",
+  "ui": "Interested in our higher-security features?",
```
Comment (julmondragon) [inline i18n/en.json:4474]: Should update this message id to something more standardized?

File: i18n/en.json [hunk 2]
```diff
@@ -3560,6 +3560,7 @@
   "leave_team_modal.yes": "Yes",
   "licensingPage.infoBanner.startTrialTitle": "Free 30 day trial!",
   "list_modal.paginatorCount": "{startCount, number} - {endCount, number} of {total, number} total",
+  "ln": "Start your free Enterprise trial now!",
```
Comment (julmondragon) [inline i18n/en.json:3563]: Should update this message id to something more standardized?

File: components/onboarding_tasklist/onboarding_tasklist_completed.tsx [hunk 3]
```diff
@@ -66,6 +77,33 @@ interface Props {
...
+            TELEMETRY_CATEGORIES.SELF_HOSTED_START_TRIAL_MODAL,
+            'open_start_trial_modal',
+        );
+        dispatch(openModal({
+            modalId: ModalIdentifiers.START_TRIAL_MODAL,
+            dialogType: StartTrialModal,
+        }));
+
+        props.dismissAction();
+    };
```
Comment (julmondragon) [inline components/onboarding_tasklist/onboarding_tasklist_completed.tsx:105]: Looks like a candidate for `useCallback`.

**Review by julmondragon — APPROVED** [review]
Looks good! Just a few comments:

File: i18n/en.json (hunk 1 above)
Comment (AGMETEOR) [reply i18n/en.json:4474]: Thanks! I forgot these

**Comment by stevemudie** [comment]
@AGMETEOR I completed all of the steps but it didn't prompt me to start the trial. If you can reset it all I'll go through it all again though to be sure.

Along the way I noticed that the link to Playbooks asks me to log in again but fails to go any further even when I log in.
Screen Shot 2022-02-21 at 2 20 58 PM
Screen Shot 2022-02-21 at 2 21 15 PM

**Comment by AGMETEOR** [comment]
@stevemudie I have set up a new one

**Review by stevemudie — APPROVED** [review]
Playbooks and boards are still a dead end but otherwise the checklist worked as expected and the Star Trial was the final prompt. Nice work @AGMETEOR!

**Pull request merged**